app/src/main/java/org/fossasia/openevent/app/core/orders/detail/SwipeController.java

```diff
@@ -0,0 +1,97 @@
+        return false;
+    }
+
+    @Override
+    public int getMovementFlags(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder) {
+        int dragFlags = 0;
+
+        if (orderDetailViewModel.getCheckedInStatus(viewHolder.getAdapterPosition()) == null)
+            makeMovementFlags(dragFlags, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT);
+
+        if (orderDetailViewModel.getCheckedInStatus(viewHolder.getAdapterPosition())) {
+            return makeMovementFlags(dragFlags, ItemTouchHelper.LEFT);
+        } else {
+            return makeMovementFlags(dragFlags, ItemTouchHelper.RIGHT);
+        }
+    }
+
+    @Override
+    public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
+        int position = viewHolder.getAdapterPosition();
+
+        orderDetailViewModel.toggleCheckIn(position);
+        orderAttendeesAdapter.notifyItemChanged(position);
+    }
+
+    @Override
+    public void onChildDraw(Canvas canvas, RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder,
+                            float dX, float dY, int actionState, boolean isCurrentlyActive) {
+
+        if (actionState == ItemTouchHelper.ACTION_STATE_SWIPE) {
+            Paint paintGreen = new Paint();
+            Paint paintRed = new Paint();
+
+            View itemView = viewHolder.itemView;
+            float height = (float) itemView.getBottom() - (float) itemView.getTop();
+            float width = height / 3;
+
+            RectF doneIconDest = new RectF((float) itemView.getLeft() + width,
+                (float) itemView.getTop() + width, (float) itemView.getLeft() + 2 * width,
+                (float) itemView.getBottom() - width);
+            RectF closeIconDest = new RectF((float) itemView.getRight() - 2 * width,
+                (float) itemView.getTop() + width, (float) itemView.getRight() - width,
+                (float) itemView.getBottom() - width);
+
+            paintGreen.setColor(context.getResources().getColor(R.color.light_green_500));
+            paintRed.setColor(context.getResources().getColor(R.color.red_500));
+
+            Bitmap doneIcon = BitmapFactory.decodeResource(context.getResources(), R.drawable.done);
+            Bitmap closeIcon = BitmapFactory.decodeResource(context.getResources(), R.drawable.close);
+
+            if (dX > 0) {
+                RectF background = new RectF((float) itemView.getLeft(), (float) itemView.getTop(), dX,
+                    (float) itemView.getBottom());
+
+                canvas.drawRect(background, paintGreen);
+
+                canvas.drawBitmap(doneIcon, null, doneIconDest, paintGreen);
+            } else {
+                RectF background = new RectF((float) itemView.getRight() + dX, (float) itemView.getTop(),
+                    (float) itemView.getRight(), (float) itemView.getBottom());
+                canvas.drawRect(background, paintRed);
+
+                canvas.drawBitmap(closeIcon, null, closeIconDest, paintRed);
```

Comment:
Last two lines from both if blocks will be moved below, and inside the function, a local paint and icon variable will be set to paintRed or paintGreen, and doneIcon or closeIcon